src/main/scala/util/TraceCoreIngressGen.scala

```diff
@@ -0,0 +1,52 @@
+// See LICENSE.Berkeley for license details.
```

Comment:
Can you make a new package for these? `src/main/scala/trace`, and `freechips.rocketchip.trace`.